tests/disaster-recovery/sc_arbiter/test_mon_osd_failures.py

```diff
@@ -0,0 +1,212 @@
+        )
+        sc_obj.get_logwriter_reader_pods(constants.LOGREADER_CEPHFS_LABEL)
+
+        wait_for_pods_to_be_in_statuses(
+            expected_statuses=constants.STATUS_COMPLETED,
+            pod_names=[pod.name for pod in sc_obj.cephfs_logreader_pods],
+            timeout=900,
+            namespace=constants.STRETCH_CLUSTER_NAMESPACE,
+        )
+        logger.info("[CephFS] Logreader job pods have reached 'Completed' state!")
+
+        assert sc_obj.check_for_data_corruption(
+            label=constants.LOGREADER_CEPHFS_LABEL
+        ), "Data is corrupted for cephFS workloads"
+        logger.info("No data corruption is seen in CephFS workloads")
+
+        assert sc_obj.check_for_data_corruption(
+            label=constants.LOGWRITER_RBD_LABEL
+        ), "Data is corrupted for RBD workloads"
+        logger.info("No data corruption is seen in RBD workloads")
+
+    request.addfinalizer(finalizer)
+
+
+@pytest.mark.usefixtures("setup_logwriter_workloads")
+class TestMonAndOSDFailures:
+    def test_single_mon_failures(self):
+        """
+        Test mon failure with IO in the background
+
+        """
+        logger.info("testing single mon failures scenario")
+        sc_obj = StretchCluster()
+
+        # get mon-pod of a single zone
+        mon_pods_in_zone = sc_obj.get_mon_pods_in_a_zone("data-1")
+        mon_pod_to_fail = random.choice(mon_pods_in_zone).name
+
+        # get the deployment of the mon-pod
+        mon_dep = get_deployment_name(mon_pod_to_fail)
+
+        # scale the deployment of mon to 0
+        # and wait 10 mins
+        logger.info(f"failing mon {mon_dep} now...")
+        if modify_deployment_replica_count(mon_dep, 0):
+            time.sleep(600)
+
+        # scale the deployment back to 1
+        logger.info(f"recovering mon {mon_dep} now...")
+        modify_deployment_replica_count(mon_dep, 1)
+
+    def test_both_mon_failure(self):
+        """
+        Test mon failure with IO for both the data-zones
+
+        """
+        logger.info("testing mon failures at both the data-zones")
+        sc_obj = StretchCluster()
+        data_zones = constants.ZONES_LABELS
+        data_zones.remove("arbiter")
+        mon_deps = list()
+        for zone in data_zones:
+            # get mon-pod of a single zone
+            mon_pods_in_zone = sc_obj.get_mon_pods_in_a_zone(zone)
+            mon_pod_to_fail = random.choice(mon_pods_in_zone).name
+
+            # get the deployment of the mon-pod
+            mon_dep = get_deployment_name(mon_pod_to_fail)
+
+            # scale the deployment of mon to 0
+            # and wait 10 mins
+            modify_deployment_replica_count(mon_dep, 0)
+            logger.info(f"Failing mon {mon_dep} from data-zone {zone}")
+            mon_deps.append(mon_dep)
+
+        time.sleep(600)
+
+        # scale the deployments back to 1
+        for mon_dep in mon_deps:
+            logger.info(f"recovering mon {mon_dep}")
```

Comment:
provide also the info that your are recovering the mon by scaling up the mon deployment